index.js

```diff
@@ -52,25 +52,48 @@ async function handleRequest(request) {
       return Response.json(
         {
           code: '404 Not Found',
-          message: 'Key does not exist or has not propagated.'
+          message: 'Key does not exist or has not propagated.',
         },
         {
-          status: 404
-        }
+          status: 404,
+        },
       )
     } else {
+      // PLAUSIBLE_HOST should be the full URL to your Plausible Analytics instance
+      // e.g. https://plausible.io/
+      if (PLAUSIBLE_HOST !== undefined) {
+        const url = PLAUSIBLE_HOST + 'api/event'
+        const headers = new Headers()
+        headers.append('User-Agent', request.headers.get('User-Agent'))
+        headers.append(
+          'X-Forwarded-For',
+          request.headers.get('X-Forwarded-For'),
+        )
+        headers.append('Content-Type', 'application/json')
+
+        const data = {
+          name: 'pageview',
+          url: request.url,
+          domain: new URL(request.url).hostname,
+        }
+        await fetch(url, {
```

Comment:
We would want to wrap this fetch (maybe the whole thing, but at least the fetch) in a waitUntil: https://developers.cloudflare.com/workers/runtime-apis/fetch-event/#waituntil
So that its forked to the background and doesn't block execution of the response and hold up the worker.